### PR TITLE
View-in-room: Select default image when "view in room" is clicked

### PR DESCRIPTION
**Pull request by lilyfromseattle**
Slider moves to default image before view in room

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit f255ade971757960c7c09954ebe65bec0305c638

https://deploy-preview-2077--artsy-reaction.netlify.com

File: src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkImageBrowser.tsx [hunk 1]
```diff
@@ -108,6 +117,15 @@ export class SmallArtworkImageBrowser extends React.Component<
...
   ArtworkBrowserState
 > {
+  slider: Slider
+
+  setSliderRef = slider => {
+    this.slider = slider
+    if (this.props.sliderRef) {
+      this.props.sliderRef(slider)
+    }
+  }
```
Comment (ds300) [inline src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkImageBrowser.tsx]: It turns out that `SmallArtworkImageBrowser` doesn't need to maintain a reference to the slider so we can delete these two things (`slider` and `setSliderRef`) and just pass `this.props.sliderRef` directly to the `ref` property on the `Slider` element.

File: src/Apps/Artwork/Components/ArtworkImageBrowser/index.tsx [hunk 2]
```diff
@@ -5,28 +5,52 @@ import { ArtworkImageBrowser_artwork } from "__generated__/ArtworkImageBrowser_a
...
+    const { images, image } = this.props.artwork
     if (!images.length) {
       return null
     }
+
+    const defaultImageIndex = images
+      .map(e => {
+        return e.id
+      })
+      .indexOf(image.id)
```
Comment (ds300) [inline src/Apps/Artwork/Components/ArtworkImageBrowser/index.tsx]: #superminor you could also do `images.findIndex(e => e.id === image.id)`

File: src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkImageBrowser.tsx [hunk 3]
```diff
@@ -17,16 +17,17 @@ import {
...
+export const ArtworkImageBrowser = (props: ArtworkBrowserProps) => {
   return (
     <>
       <Media at="xs">
-        <SmallArtworkImageBrowser images={props.images} />
+        <SmallArtworkImageBrowser {...props} />
       </Media>
       <Media greaterThan="xs">
-        <LargeArtworkImageBrowser images={props.images} />
+        <LargeArtworkImageBrowser {...props} />
```
Comment (damassi) [inline src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkImageBrowser.tsx]: Why use the spread here?

Comment (ds300) [reply src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkImageBrowser.tsx]: These two components take the same props as the wrapper component, so why not pass them all along.

**Comment by sweir27** [comment]
Agree with comments above ☝️ but otherwise looks good to me! ✨ 

Would you mind adding a screen cap (I use `LiceCAP` but there are probably other, fancier solutions) that shows the new behavior?

In this case, since all this PR does is explicitly select the default image when "View in room" is clicked and sets the `data-is-default` attribute, can we update the description to reflect that? Maybe something like `View-in-room: Select default image when "view in room" is clicked`?

**Comment by ds300** [comment]
> other, fancier solutions

https://getkap.co/ :heart:

**Comment by artsyit** [comment]
:rocket: PR was released in v14.1.28 :rocket:

**Pull request merged**